src/items/weapons/weapons.cpp

```diff
@@ -592,7 +604,50 @@ bool WeaponMelee::useWeapon(Player* player, Item* item, Creature* target) const
 	if (damageModifier == 0) {
 		return false;
 	}
+	if (player->getCleavePercent() > 0) {
+		const Position& targetPos = target->getPosition();
+		const Position& playerPos = player->getPosition();
+		if (playerPos != targetPos) {
+			Position firstCleaveTargetPos = targetPos;
+			Position secondCleaveTargetPos = targetPos;
+			if (targetPos.x == playerPos.x) {
+				firstCleaveTargetPos.x++;
+				secondCleaveTargetPos.x--;
+			} else if (targetPos.y == playerPos.y) {
+				firstCleaveTargetPos.y++;
+				secondCleaveTargetPos.y--;
+			} else {
+				if (targetPos.x > playerPos.x)
+					firstCleaveTargetPos.x--;
+				else
+					firstCleaveTargetPos.x++;
+
+				if (targetPos.y > playerPos.y)
+					secondCleaveTargetPos.y--;
+				else
+					secondCleaveTargetPos.y++;
+			}
+			Position pos(firstCleaveTargetPos.x, firstCleaveTargetPos.y, firstCleaveTargetPos.z);
+			Position pos2(secondCleaveTargetPos.x, secondCleaveTargetPos.y, secondCleaveTargetPos.z);
+			Tile* firstTile = g_game().map.getTile(pos);
+			Tile* secondTile = g_game().map.getTile(pos2);
+
+			if (firstTile) {
```

Comment:
```suggestion
			if (firstTile && !firstTile->isMoveableBlocking()) {
```
this fixed wall crash for me